### PR TITLE
fix(cli): render query results from recall response

### PR DESCRIPTION
**Pull request by joshuaswarren**
## Summary
- render non-JSON `remnic query` output from `recall().results` instead of the non-existent `memories` field
- add a small renderer covering content, preview/context fallback, and no-results output

## Verification
- `pnpm exec tsx --test packages/remnic-cli/src/query-render.test.ts`
- `pnpm --filter @remnic/cli check-types`
- `pnpm --filter @remnic/cli test`
- `npm run preflight:quick`

Clawpatch finding: `medium: Plain-text query output reads a non-existent recall field`.

---

> [!NOTE]
> **Low Risk**
> Low risk: scoped to `remnic query` CLI formatting and recall request metadata, with added unit tests to lock output behavior.
> 
> **Overview**
> Fixes non-JSON `remnic query` output to render from `recall().results` (with `content` → `preview` → per-item `context` fallbacks) instead of a non-existent `memories` field, and returns "No results." when empty.
> 
> Adds `buildQueryRecallRequest()` to include a per-process `sessionKey` in recall calls (used for both normal and `--explain` paths) and introduces `query-render.test.ts` to cover request shape and text rendering edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f483c720cd0a2a920ddcaa3dfec91432250e6b53. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by joshuaswarren** [comment]
@cursor review

File: packages/remnic-cli/src/index.ts [hunk 1]
```diff
@@ -3845,6 +3845,30 @@ async function cmdStatus(json: boolean): Promise<void> {
...
+}): string[] {
+  const results = Array.isArray(result.results) ? result.results : [];
+  if (results.length === 0) return ["No results."];
+
+  return results.map((memory) => {
+    const text =
+      memory.content?.trim()
+      || memory.preview?.trim()
+      || memory.context?.trim()
+      || result.context?.trim()
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-cli/src/index.ts]: Fixed in 1a97c496 by removing the aggregate recall context fallback from per-item text rendering and adding a regression test that keeps aggregate context out of bullet output.

**Comment by joshuaswarren** [comment]
Addressed the Codex session-key finding in f483c720: `remnic query` now builds a one-shot CLI recall request with a `sessionKey` and uses it for normal output plus fallback explain recall, with a regression test for the request shape.

**Comment by joshuaswarren** [comment]
@cursor review
